app/commands/v2/discard_draft.rb

```diff
@@ -11,6 +11,8 @@ def call
           delete_draft
         end
 
+        delete_access_limit
```

Comment:
Could potentially be moved before the update/delete of the draft. Seems more intuitive that way.